### PR TITLE
Admin: allow direct link to relevant admin page from ecoverse / challenge / opportunity

### PR DESCRIPTION
**Pull request by atstojanov**
For returning back from admin panel can be used the entity display name on the top of the settings page. It is a link. Open to suggestions how to improve it.

### CODE REVIEW

File: src/components/Admin/ManagementPageTemplate.tsx [hunk 1]
```diff
@@ -17,16 +17,33 @@ interface Props extends PageProps {
...
   useUpdateNavigation({ currentPaths: paths });
 
+  const titleElement = useMemo(
+    () =>
+      returnUrl ? (
+        <Link component={RouterLink} to={returnUrl}>
+          {title}
+        </Link>
+      ) : (
+        title
```
Comment (hero101) [inline src/components/Admin/ManagementPageTemplate.tsx:36]: What is the point of having just the text?

File: src/components/Admin/ManagementPageTemplate.tsx [hunk 2]
```diff
@@ -17,16 +17,33 @@ interface Props extends PageProps {
       url: string;
     }>;
   }>;
+  title?: string;
+  returnUrl?: string;
```
Comment (hero101) [inline src/components/Admin/ManagementPageTemplate.tsx:21]: This must represent that it's not actually a return URL, but rather the entity's page URL imo.

File: src/pages/Ecoverse.tsx [hunk 3]
```diff
@@ -197,7 +203,20 @@ const EcoversePage: FC<EcoversePageProps> = ({ paths, ecoverse }): React.ReactEl
...
+            permissions.edit && (
+              <SettingsButton
+                color={'primary'}
+                to={`/admin/ecoverses/${ecoverseId}`}
+                tooltip={t('pages.ecoverse.sections.header.buttons.edit.tooltip')}
+              />
+            )
+          }
+        >
+          {/* {user?.isAdmin && <Box display="fixed" top={0}></Box>} */}
```
Comment (hero101) [inline src/pages/Ecoverse.tsx:218]: remove

File: src/pages/Ecoverse.tsx [hunk 4]
```diff
@@ -51,9 +50,16 @@ const useStyles = createStyles(theme => ({
 
 interface EcoversePageProps extends PageProps {
   ecoverse: EcoverseInfoQuery;
+  permissions: {
```
Comment (hero101) [inline src/pages/Ecoverse.tsx:53]: If all the pages use the `permissions` field, it's better to have it in `PageProps` or extend it from a different interface.

File: src/routing/admin/challenge/challenge.tsx [hunk 5]
```diff
@@ -71,7 +71,12 @@ const ChallengeRoutes: FC<PageProps> = ({ paths }) => {
   return (
     <Switch>
       <Route exact path={`${path}`}>
-        <ManagementPageTemplate data={managementData.challengeLvl} paths={currentPaths} />
+        <ManagementPageTemplate
+          data={managementData.challengeLvl}
+          paths={currentPaths}
+          title={data?.ecoverse.challenge.displayName}
+          returnUrl={`/${ecoverseId}/challenges/${challengeId}`}
```
Comment (hero101) [inline src/routing/admin/challenge/challenge.tsx:78]: I think it's a good idea to use the tools from `utils\urlBuilders`

File: src/routing/admin/ecoverse/ecoverse.tsx [hunk 6]
```diff
@@ -67,7 +67,12 @@ export const EcoverseAdminRoute: FC<EcoverseAdminRouteProps> = ({ paths }) => {
   return (
     <Switch>
       <Route exact path={`${path}`}>
-        <ManagementPageTemplate data={managementData.ecoverseLvl} paths={currentPaths} />
+        <ManagementPageTemplate
+          data={managementData.ecoverseLvl}
+          paths={currentPaths}
+          title={ecoverse?.ecoverse.displayName}
+          returnUrl={`/${ecoverseId}`}
```
Comment (hero101) [inline src/routing/admin/ecoverse/ecoverse.tsx:74]: `utils\urlBuilders`

File: src/routing/admin/opportunity/opportunity.tsx [hunk 7]
```diff
@@ -77,7 +77,12 @@ export const OpportunityRoutes: FC<Props> = ({ paths, challengeId: challengeUUID
   return (
     <Switch>
       <Route exact path={`${path}`}>
-        <ManagementPageTemplate data={managementData.opportunityLvl} paths={currentPaths} />
+        <ManagementPageTemplate
+          data={managementData.opportunityLvl}
+          paths={currentPaths}
+          title={data?.ecoverse.opportunity.displayName}
+          returnUrl={`/${ecoverseId}/challenges/${challengeId}/opportunities/${opportunityId}`}
```
Comment (hero101) [inline src/routing/admin/opportunity/opportunity.tsx:84]: `utils\urlBuilders`

File: src/routing/ecoverse.tsx [hunk 8]
```diff
@@ -49,7 +58,7 @@ export const EcoverseRoute: FC<PageProps> = ({ paths }) => {
   return (
     <Switch>
       <Route exact path={path}>
-        {!loading && <EcoversePage ecoverse={ecoverseInfo} paths={currentPaths} />}
+        {!loading && <EcoversePage ecoverse={ecoverseInfo} paths={currentPaths} permissions={{ edit: isAdmin }} />}
```
Comment (hero101) [inline src/routing/ecoverse.tsx:61]: I think the loading check is not necessary here.

File: src/routing/ecoverse.tsx [hunk 9]
```diff
@@ -106,7 +126,9 @@ const Challenge: FC<ChallengeRootProps> = ({ paths, challenges }) => {
         <Opportunity opportunities={challenge.opportunities} paths={currentPaths} challengeUUID={challenge.id} />
       </Route>
       <Route exact path={path}>
-        {!loading && <ChallengePage challenge={challenge as ChallengeType} paths={currentPaths} />}
+        {!loading && (
```
Comment (hero101) [inline src/routing/ecoverse.tsx:129]: unnecessary loading check imo

File: quickstart-services.yml [hunk 10]
```diff
@@ -86,7 +86,7 @@ services:
       - 4456
     environment:
       - LOG_LEVEL=debug
-    restart: on-failure
+    restart: unless-stopped
```
Comment (hero101) [inline quickstart-services.yml:89]: @valentinyanakiev can you verify this one?

**Review by hero101 — CHANGES_REQUESTED** [review]

File: src/pages/Ecoverse.tsx [hunk 11]
```diff
@@ -197,7 +203,20 @@ const EcoversePage: FC<EcoversePageProps> = ({ paths, ecoverse }): React.ReactEl
           />
         }
       >
-        <SectionHeader text={name} />
+        <SectionHeader
+          text={name}
+          editComponent={
+            permissions.edit && (
+              <SettingsButton
+                color={'primary'}
```
Comment (hero101) [inline src/pages/Ecoverse.tsx:211]: The color here is `primary` but on the other pages, it's the default.
Also probably a bigger icon would look better.

**Review by hero101 — CHANGES_REQUESTED** [review]

File: src/components/Admin/ManagementPageTemplate.tsx (hunk 1 above)
Comment (atstojanov) [reply src/components/Admin/ManagementPageTemplate.tsx:36]: If the link is missing the "Title" still can be shown.

Comment (hero101) [reply src/components/Admin/ManagementPageTemplate.tsx:36]: That was exactly my point. Why do we need to show the title there?

File: src/pages/Ecoverse.tsx (hunk 4 above)
Comment (hero101) [reply src/pages/Ecoverse.tsx:53]: Can it be defined as `permissions: Record<string, boolean>`?

**Comment by techsmyth** [comment]
Looks like there is a strange effect in the route when drill down into opportunity:

"http://localhost:3000/admin/ecoverses/un_sdgs/challenges/undefined/opportunities/food-sec" - see the undefined. This is then causing problems when navigating back to the opportunity.

File: quickstart-services.yml (hunk 10 above)
Comment (atstojanov) [reply quickstart-services.yml:89]: I'll move that in separate PR.

File: src/pages/Ecoverse.tsx (hunk 11 above)
Comment (atstojanov) [reply src/pages/Ecoverse.tsx:211]: The Ecoverse page does not have banner and because of that it requires different color.

Comment (hero101) [reply src/pages/Ecoverse.tsx:211]: I think those colors have a meaning and are not there just for styling reasons.

Comment (techsmyth) [reply src/pages/Ecoverse.tsx:211]: Suggest we keep it as it is now and pick this up as part of the wider UX review / update....

File: src/components/Admin/ManagementPageTemplate.tsx (hunk 1 above)
Comment (atstojanov) [reply src/components/Admin/ManagementPageTemplate.tsx:36]: It makes the settings page more complete

File: src/pages/Challenge.tsx [hunk 12]
```diff
@@ -227,31 +220,15 @@ const Challenge: FC<ChallengePageProps> = ({ paths, challenge }): React.ReactEle
...
-                    color={'white'}
-                    width={20}
-                    height={20}
-                    className={styles.edit}
-                    onClick={() => setIsEditOpened(true)}
+              editComponent={
+                permissions.edit && (
+                  <SettingsButton
+                    to={`/admin/ecoverses/${ecoverseId}/challenges/${challenge.nameID}`}
+                    tooltip={t('pages.challenge.sections.header.buttons.edit.tooltip')}
```
[the remark this answers is not in the document]
Comment (atstojanov) [reply src/pages/Challenge.tsx]: changed to "<Entity> Settings"

File: src/pages/Ecoverse.tsx (hunk 11 above)
Comment (atstojanov) [reply src/pages/Ecoverse.tsx:212]: @techsmyth Agree

File: src/pages/Ecoverse.tsx (hunk 4 above)
Comment (atstojanov) [reply src/pages/Ecoverse.tsx:54]: defining it as Record<string, boolean> makes it not specific enough. Better global `Authorization` functionality is required.

**Review by hero101 — APPROVED** [review]

**Pull request merged**